ledger-api/grpc-definitions/com/daml/ledger/api/v2/participant_offset.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+
+option java_outer_classname = "ParticipantOffsetOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// Describes a specific point on the participant. This is a participant local value:
+// changes synchronized via a specific domain should have different participant offsets
+// on different participants.
+//
+// This is also as a unique index of the changes which happened on the virtual shared ledger.
+// Participant offset define an order, which is the same in which order the updates are
+// visible as subscribing to the ``UpdateService``. This ordering is also a fully causal
+// ordering for one specific domain: for two updates synchronized by the same domain, the
+// one with a bigger participant offset happened after than the one with a smaller participant
+// offset. Please note this is not true for updates synchronized by a different domain.
+//
+// The Ledger API endpoints that take offsets allow to specify portions
+// of the participant that are relevant for the client to read.
+//
+// Offsets returned by the Ledger API can be used as-is (e.g.
+// to keep track of processed transactions and provide a restart
+// point to use in case of need).
+//
+// The format of absolute offsets is opaque to the client: no
+// client-side transformation of an offset is guaranteed
+// to return a meaningful offset.
+//
+// The server implementation ensures internally that offsets
+// are lexicographically comparable.
+message ParticipantOffset {
+
+  oneof value {
+    // The format of this string is specific to the participant and opaque to the client.
+    string absolute = 1;
+    ParticipantBoundary boundary = 2;
+  }
+
+  enum ParticipantBoundary {
+    // Refers to the first transaction.
```

Comment:
Do you want to slot in a "currently" like for end, as pruning can move the "first" and it's not clear that there is no pruning happen concurrently.

Reply:
It seems to me from the implementation that this is actually fetching from the beginning...this also meas that this is not usable for stream start after the first pruning.
I am not sure this is practical, but I would not change the doc here, because the current implementation is aligned with the current doc.